python/ray/experimental/signal.py

```diff
@@ -0,0 +1,168 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import hashlib
+import time
+
+import ray
+
+SYSTEM_ID_STRING = ray.ray_constants.ID_SIZE * b"\xaa"
+SYSTEM_ID = ray.ObjectID(SYSTEM_ID_STRING)
+
+SIG_USER = 100
+
+class Signal(object):
+    """Signal object"""
+    def __init__(self, type, value):
+        self.sig_type = type
+        self.sig_value = value
+    def value(self):
+        return self.sig_value
+    def type(self):
+        return self.sig_type
+
+def _get_signal_object_id_from_binary(source_id, counter):
+    m = hashlib.sha1()
```

Comment:
Change this function so that `source_id` is a task ID. Implement a Python extension for `ComputeReturnId`, then use `ComputeReturnId(task_id, counter)` to compute the signal object ID.